test/nn/test_init.py

```diff
@@ -121,6 +121,15 @@ def test_trunc_normal(self):
 
             assert self._is_trunc_normal(input_tensor, mean, std, a, b)
 
+    def test_trunc_normal_generator(self):
+        gen = torch.Generator()
+        gen.manual_seed(42)
+        input_tensor = torch.empty(5)
+        init.trunc_normal_(input_tensor, generator=gen)
+        expected = torch.tensor([1.1025, 1.2598, -0.2840, 1.5415, -0.2652])
```

Comment:
Nothing gurantees that the rng won't change across versions.
You can use `self._is_trunc_normal()` on the generated Tensor to ensure it was properly generated (see test above for details).

You also want to make sure the generator is being used properly, you can add:
```python
ref = torch.empty(5)
torch.manual_seed(42)
init.trun_normal_(ref)

self.assertEqual(input_tensor, ref)
```

Reply:
Just fixed this -- my apologies as I didn't thoroughly check how self._is_trunc_normal() was implemented (I incorrectly assumed it used an equivalent method in scipy, and I wasn't sure if scipy had generator support). I'll be sure to take a closer look at surrounding tests in the future, hope this is good now!

Reply:
No worries. Thanks for the quick update!